Coding/fw/lib/Lawicel/CANInterface.h

```diff
@@ -0,0 +1,157 @@
+/***************************************************************************************************
+  (c) NewTec GmbH 2020   -   www.newtec.de
+  $URL: https://github.com/NewTec-GmbH/esp32-can-iot $
+***************************************************************************************************/
+/**
+@addtogroup Interfaces
+@ingroup Lawicel
+@{
+@file       CANInterface.h
+
+CAN interface for Lawicel Protocol
+
+* @}
+***************************************************************************************************/
+#ifndef CAN_INTERFACE_H
+#define CAN_INTERFACE_H
+
+/* INCLUDES ***************************************************************************************/
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* FORWARD DECLARATIONS ***************************************************************************/
+
+class CANInterface
+{
+public:
+    /* CONSTANTS ******************************************************************************/
+
+    /* TYPES **********************************************************************************/
+
+    /** Enum of Possible States of the CAN Channel
+     */
+    enum BUS_STATE
+    {
+        CLOSED = 0,
+        NORMAL,
+        LISTEN_ONLY
+    };
+
+    /**
+    * Format of a CAN Frame with an empty constructor
+    */
+    struct Frame
+    {
+        uint32_t ID;     //CAN ID
+        bool RTR;        //Identifies a RTR Frame
+        bool Extended;   //Identifies an Extended Frame
+        uint8_t DLC;     //Data Length
+        uint8_t Data[8]; //Data of the Frame
+
+        Frame() : ID(0),
+                  RTR(false),
+                  Extended(false),
+                  DLC(0)
+        {
+            for (int i = 0; i < 8; i++)
+            {
+                Data[i] = 0;
+            }
+        }
+    };
+
+    /**
+    * Default constructor.
+    */
+    CANInterface()
+    {
+    }
+
+    /**
+    * Default destructor.
+    */
+    virtual ~CANInterface()
+    {
+    }
+
+    /** 
+    * Initialize Module.
+    */
+    virtual uint8_t begin() = 0;
+
+    /** 
+    * Terminate Module. 
+    */
+    virtual uint8_t end() = 0;
+
+    /**
+    * Send a Data String.
+    * @param[in] Frame         Frame to be sended via CAN BUS
+    */
+    virtual uint8_t send(const Frame &Frame) = 0;
+
+    /**
+    * Set the State of the CAN Channel.
+    * @param[in] state         Channel State to be set
+    */
+    virtual uint8_t setState(const BUS_STATE state) = 0;
```

Comment:
const is usually only applied to pointer and references.  This variable is pushed by value, meaning the function gets a private copy. Forcing it to const is a little over restriktive.